crengine/src/lvtinydom.cpp

```diff
@@ -6406,9 +6406,11 @@ int initTableRendMethods( ldomNode * enode, int state )
 
 bool hasInvisibleParent( ldomNode * node )
 {
-    for ( ; !node->isRoot(); node = node->getParentNode() )
-        if ( node->getStyle()->display==css_d_none )
+    for ( ; node && !node->isRoot(); node = node->getParentNode() ) {
+        css_style_ref_t style = node->getStyle();
+        if (!style.isNull() && style->display == css_d_none)
```

Comment:
Got a crash? Or just some reports from code analysis tools?
Because unless I missed something with the call from the added MathML bit, the only other call is from initNodeRendMethod, where there is always a style defined at this point.
And when walking nodes up, we should always meet the root node (so, !node->isRoot() will get us out of this loop), and so we should never meet a NULL node.

Reply:
> Got a crash?

Yes, of cource.


> Because unless I missed something with the call from the added MathML bit, the only other call is from initNodeRendMethod, where there is always a style defined at this point.

Crashed when style is null ref.

> And when walking nodes up, we should always meet the root node (so, !node->isRoot() will get us out of this loop), and so we should never meet a NULL node.

I just added it just in case, it costs absolutely nothing.

Reply:
> Crashed when style is null ref.

Can you get a backtrace? I'd like to see from which of the 2 hasInvisibleParent() calls this happen.

Reply:
> Can you get a backtrace? I'd like to see from which of the 2 hasInvisibleParent() calls this happen.

Sorry, I lost it. Document the same: http://fred-wang.github.io/mathml.css/